### PR TITLE
Fix post deployment so that it shows even if the bundle is modified.

### LINKED ISSUE #3772: getstarted not shown when adding additional apps
To dupe:
- deploy cdk;
- deploy aws integrator;
- commit;
- no getstarted is shown on the right hand side.

We should persist the information that a bundle is deployed in the model. We decided that every application composing a bundle is annotated with the bundle, and this is supposedly in place already. Then we should check if a bundle is (or was) deployed in the model, by looking at all app annotations, and:
- open the getstarted pane after deployment;
- allow for reopening it even after a refresh.

### PR DESCRIPTION
**Pull request by hatched**
Fixes #3772

### CODE REVIEW

**Comment by hatched** [comment]
#### To QA

Add CDK to the canvas and then add AWS-Integrator and relate them.
Deploy, once the deployment flow closes you should see the post deployment panel.

File: jujugui/static/gui/src/app/components/post-deployment/post-deployment.js [hunk 1]
```diff
@@ -35,106 +35,48 @@ class PostDeployment extends React.Component {
...
-      this.setState({
-        content: markdown
-      });
+    if (fileContents.substring(0, 1) === '{') {
+      return;
     }
+    if (type === 'content') {
+      fileContents = this._processGetStarted(fileContents);
+    }
+    this.setState({[type]: fileContents});
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/components/post-deployment/post-deployment.js]: What happens if there are multiple `entityURLs`?

File: jujugui/static/gui/src/app/init/component-renderers-mixin.js [hunk 2]
```diff
@@ -523,31 +520,6 @@ Browser: ${navigator.userAgent}`
...
-      // Round it to the nearest second.
-      let timeOpen = Math.round(
-        (closeTime - openTime) / 1000
-      );
-      let args = [
-        `${timeOpen}s`,
-        entityId
-      ];
-
-      this.sendAnalytics(
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/init/component-renderers-mixin.js]: Do we not care about these analytics anymore?

File: jujugui/static/gui/src/app/components/post-deployment/post-deployment.js [hunk 3]
```diff
@@ -35,106 +35,48 @@ class PostDeployment extends React.Component {
...
-        this.replaceTemplateTags(
-          frontmatterAndMarkdown.markdown
-        ),
-        {renderer: renderer}
-      );
-
-      this.setState({
-        content: markdown
-      });
+    if (fileContents.substring(0, 1) === '{') {
```
Comment (frankban) [inline jujugui/static/gui/src/app/components/post-deployment/post-deployment.js]: This line really deserves a comment.

File: jujugui/static/gui/src/app/components/post-deployment/post-deployment.js [hunk 4]
```diff
@@ -249,7 +189,7 @@ class PostDeployment extends React.Component {
     Show the details page of a charm or bundle based on the click.
   */
   _handleViewDetails() {
-    const url = urls.URL.fromAnyString(this.props.entityId);
+    const url = urls.URL.fromAnyString(this.props.entityURLs[0]);
```
[the remark this answers is not in the document]
Comment (frankban) [reply jujugui/static/gui/src/app/components/post-deployment/post-deployment.js]: This is an initial step, later we'll handle multiple getstarted.

File: jujugui/static/gui/src/app/init/bundle-importer.js [hunk 5]
```diff
@@ -130,13 +130,13 @@ class BundleImporter {
...
-        // then add the bundle-url annotation to it.
+        // then add the bundleURL annotation to it.
         const found = changes.some(innerRecord => {
           if (innerRecord.method === 'setAnnotations') {
             if (innerRecord.args[0] === `$${record.id}`) {
               // If this setAnnotations is for the outer record then add the
-              // bundle-url key to it.
-              innerRecord.args[2]['bundle-url'] = bundleURL;
+              // bundleURL key to it.
+              innerRecord.args[2]['bundleURL'] = bundleURL;
```
Comment (frankban) [inline jujugui/static/gui/src/app/init/bundle-importer.js]: `innerRecord.args[2].bundleURL` ?

**Comment by frankban** [comment]
QA is ok!

File: jujugui/static/gui/src/app/components/post-deployment/post-deployment.js [hunk 6]
```diff
@@ -35,106 +35,48 @@ class PostDeployment extends React.Component {
...
   }
 
-  /**
-    Callback handler for getting the entity details.
-
-    @param {Object} error Error from the API.
-    @param {Array} entityData The returned data.
-  */
-  _getEntityCallback(error, entityData) {
+  _handleFileResponse(type, error, fileContents) {
```
[the remark this answers is not in the document]
Comment (hatched) [reply jujugui/static/gui/src/app/components/post-deployment/post-deployment.js]: done

File: jujugui/static/gui/src/app/components/post-deployment/post-deployment.js (hunk 3 above)
Comment (hatched) [reply jujugui/static/gui/src/app/components/post-deployment/post-deployment.js]: The comment is there still....just above all this red :)

File: jujugui/static/gui/src/app/components/post-deployment/post-deployment.js (hunk 1 above)
Comment (hatched) [reply jujugui/static/gui/src/app/components/post-deployment/post-deployment.js]: it does....this isn't the final solution but an interim one until we have a real plan for multiple different versions of them.

File: jujugui/static/gui/src/app/components/post-deployment/post-deployment.js [hunk 7]
```diff
@@ -35,106 +35,48 @@ class PostDeployment extends React.Component {
...
-
-    @param {String} error Error from the API.
-    @param {String} postDeploymentScript The content of the script.
-  */
-  _getPostDeploymentScriptCallback(error, postDeploymentScript) {
-    if (error) {
-      console.error(error);
-      return;
-    }
+  _processGetStarted(fileContents) {
```
[the remark this answers is not in the document]
Comment (hatched) [reply jujugui/static/gui/src/app/components/post-deployment/post-deployment.js]: done

File: jujugui/static/gui/src/app/init/bundle-importer.js (hunk 5 above)
Comment (hatched) [reply jujugui/static/gui/src/app/init/bundle-importer.js]: done

File: jujugui/static/gui/src/app/init/component-renderers-mixin.js (hunk 2 above)
Comment (hatched) [reply jujugui/static/gui/src/app/init/component-renderers-mixin.js]: Not at the moment, we need to find a better way of tracking these.

**Pull request merged**